Neos.Neos/Classes/Controller/Module/Administration/UsersController.php

```diff
@@ -139,8 +161,14 @@ public function newAction(User $user = null)
      */
     public function createAction($username, array $password, User $user, array $roleIdentifiers, $authenticationProviderName = null)
     {
-        $this->userService->addUser($username, $password[0], $user, $roleIdentifiers, $authenticationProviderName);
-        $this->addFlashMessage('The user "%s" has been created.', 'User created', Message::SEVERITY_OK, [htmlspecialchars($username)], 1416225561);
+        $currentUserRoles = $this->userService->getAllRoles($this->currentUser);
+        $isCreationAllowed = count(array_diff($roleIdentifiers, $currentUserRoles)) === 0;
```

Comment:
I guess, that will not work. That only allows creating users, which has exact the same roles then I have.